src/Controller/Frontend.php

```diff
@@ -443,6 +443,16 @@ public function template($template)
         return $this->render($template);
     }
 
+    public function isHome($content)
```

Comment:
In before @CarsonF :wink: … But shouldn't this either be private or somewhere else. It just feels like more of the historical disasters we're trying to get away from.

Reply:
Where do you reckon? It was in the actual content record before, which is a bit of a nightmare because we would have to inject all the config into the content entities.

That info is all available in the controller and it seems like the sort of thing that can/should be done there because you're using that info to decide which template to show etc.

Reply:
Yeah, of everything I commented on here, this was the one that made my stomach churn as it is really hard … as above, my single worry is just public API (ab)use and the lack of flexibility in changing if we need to. If it is a helper it I'd guess a protected method in `Base`? … Please see my comment as looking to achieve a consensus with you too

Reply:
cool, that sounds fair enough. 

Only other strategy I can think of since the goal of this is to add some metadata to the request object is to do something with an event that can have the selected record passed in along with the `$app['config']` and `$app['request']` services passed in as dependencies.

If we did that though it would probably have to be up to the controller to trigger the event, unless we already have an event we could hook into.

Reply:
>  since the goal of this is to add some metadata to the request object is to do something with an event 

Actually, that's a bloody good idea! 

I'm going to get slapped for the suggestion of location, but `Bolt\EventListener\GeneralListener` has to have the container passed due to needing to not construct `Config` early, so adding `KernelEvents::REQUEST => 'onRequest',` to the `getSubscribedEvents` would be totally feasible